src/ucp/core/ucp_context.h

```diff
@@ -39,6 +40,30 @@ UCP_UINT_TYPE(UCP_MAX_LANES)         ucp_lane_map_t;
 #define UCP_MD_LANE_MAP_BITS         64 /* should be UCP_MD_INDEX_BITS * UCP_MAX_LANES */
 UCP_UINT_TYPE(UCP_MD_LANE_MAP_BITS)  ucp_md_lane_map_t;
 
+#if UCP_ENABLE_MT
+#  define UCP_CONTEXT_THREAD_CS_ENTER(_context) {                       \
+        if ((_context)->config.thread_mode == UCS_THREAD_MODE_MULTI) {  \
+            if (!(_context)->config.ext.use_mt_mutex) {                 \
+                ucs_spin_lock((_context)->thread_spinlock);             \
```

Comment:
I think this should be a pointer ( ucs_spin_lock(&(_context)->thread_spinlock); )